py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,314 @@
+                {
+                    "placeholder_formats": {
+                        "density_level": ":.0f",
+                        "max_bike_overflow": ":.0f",
+                        "nb_bike": ":.0f",
+                        "nb_bike_overflow": ":.0f",
+                        "nb_dock": ":.0f",
+                        "nb_e_bike_overflow": ":.0f",
+                        "nb_e_dock": ":.0f",
+                        "nb_ebike": ":.0f",
+                        "nb_free_dock": ":.0f",
+                        "nb_free_e_dock": ":.0f",
+                        "station_gps_latitude": ":.3f",
+                        "station_gps_longitude": ".3f",
+                        "station_due_date": ".0f",
+                    },
+                    "format_strings": ["format_station"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        if not self.station_codes:
+            raise Exception(STRING_MISSING_STATIONS)
+
+        # string to list if necessary
+        if not isinstance(self.station_codes, list):
+            self.station_codes = [self.station_codes]
+
+        # take the whole map for first run
+        # default values take all stations
+        # around Paris
+        # Area is then shrink to only get
+        # specified stations
+        self.gps_dict = {
+            "gpsTopLatitude": 49.0,
+            "gpsTopLongitude": 2.5,
+            "gpsBotLatitude": 48.6,
+            "gpsBotLongitude": 2.2,
+            "zoomLevel": 15,
+        }
+        self.toggled = False
+        self.idle_time = 0
+        self.request_timeout = 10
+        self.station_states = {"Operative": 0, "Close": 1}
+        self.station_index = 1
+        self.stations = {}
+        self.optimal_area = False
+        self.first_start = True
+        self.button_refresh = 2
+
+        # get placeholders
+        self.placeholders = []
+        for x in [self.format, self.format_station]:
+            self.placeholders += self.py3.get_placeholders_list(x)
+        self.placeholders += ["station_gps_latitude", "station_gps_longitude"]
+
+    def _camel_to_snake_case(self, data):
+        if not isinstance(data, (int, float)):
+            s1 = sub("(.)([A-Z][a-z]+)", r"\1_\2", data)
+            return sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+        else:
+            return data
+
+    def _cast_number(self, value):
+        try:
+            value = float(value)
+        except ValueError:
+            try:
+                value = int(value)
+            except ValueError:
+                pass
+        return value
+
+    def _set_optimal_area(self, data):
+        """
+        reduce the zone to reduce the size of fetched data on refresh
+        """
+
+        latitudes = []
+        longitudes = []
+
+        for x in data:
+            latitudes.append(float(data[x]["station_gps_latitude"]))
+            longitudes.append(float(data[x]["station_gps_longitude"]))
```

Comment:
(Suggestion) You probably can reduce 7 lines here to 2 lines by using list comprehension or maybe map.

Reply:
done 2 lines from 6 with simple list comprehension